### PR TITLE
configurable ffmpeg timeout

### PR DESCRIPTION
**Pull request by spacebares**
With the default hard-coded 10 seconds, there is a chance to lose a whole 10 seconds of footage when a **battery camera** activates.
If you configure the timeout to 1 second, now the maximum amount of footage lost is 1 second of footage.
_Edit: forgot to mention, of course there is ffmpeg negotiation time when opening rtsp stream. For me my machine takes 2-3 extra seconds for this._

This can increase the rate of which ffmpeg processes spawn by 10x but in my testing none of these processes linger even after months of operation and consume almost no cpu when spawned.

Obviously not the best solution, but hardcoding 10 seconds isnt either 😉

### CODE REVIEW

File: frigate/config.py [hunk 1]
```diff
@@ -463,6 +463,9 @@ class FfmpegConfig(FrigateBaseModel):
         default_factory=FfmpegOutputArgsConfig,
         title="FFmpeg output arguments per role.",
     )
+    timeout: float = Field(
```
Comment (NickM-27) [inline frigate/config.py:466]: I don't think timeout is a good name as it's confused with the actual meaning of timeout (how long of no communication should result in an error).

I think `healthcheck_interval` might be better

File: frigate/video.py [hunk 2]
```diff
@@ -251,8 +251,9 @@ def run(self):
                 }
             )
 
-        time.sleep(10)
-        while not self.stop_event.wait(10):
+        sleeptime = self.config.ffmpeg.timeout
```
Comment (NickM-27) [inline frigate/video.py:254]: There's no reason to read from the config every time, should be set as self.healthcheck_interval in init and then read here.

**Review by NickM-27 — CHANGES_REQUESTED** [review]

**Review by NickM-27 — APPROVED** [review]

**Comment by blakeblackshear** [comment]
I actually think `retry_interval` or `reconnect_interval` is a more appropriate name. Healthcheck implies a check against a running service.

**Comment by spacebares** [comment]
maybe `errorcheck_interval` or `errorscan_interval` ?

**Comment by NickM-27** [comment]
Right, I think for the functionality retry_interval makes sense to me

File: frigate/config.py [hunk 3]
```diff
@@ -463,6 +463,9 @@ class FfmpegConfig(FrigateBaseModel):
         default_factory=FfmpegOutputArgsConfig,
         title="FFmpeg output arguments per role.",
     )
+    retry_interval: float = Field(
+        default=10.0, title="Time in seconds in-between FFmpeg health checks."
```
Comment (blakeblackshear) [inline frigate/config.py:467]:
```suggestion
        default=10.0, title="Time in seconds to wait before FFmpeg retries connecting to the camera."
```

**Comment by blakeblackshear** [comment]
This looks good. It just needs to be added to the configuration reference in the docs.

**Review by NickM-27 — APPROVED** [review]

**Comment by NickM-27** [comment]
oops looks like black needs to be run

**Pull request merged**